src/main/java/pl/numerki/frontend/ChartGenerator.java

```diff
@@ -0,0 +1,81 @@
+package pl.numerki.frontend;
+
+import org.jfree.chart.ChartFactory;
+import org.jfree.chart.JFreeChart;
+import org.jfree.chart.plot.PlotOrientation;
+import org.jfree.chart.plot.XYPlot;
+import org.jfree.chart.renderer.xy.XYLineAndShapeRenderer;
+import org.jfree.data.xy.XYSeries;
+import org.jfree.data.xy.XYSeriesCollection;
+
+import java.awt.*;
+import java.util.function.Function;
+
+public class ChartGenerator {
+    public static JFreeChart generatePlot(
+            Function<Double, Double> function, double leftCompartment, double rightCompartment,
+            double bisectionZeroPosition, double secantZeroPosition, String chartName
+    ) {
+        XYSeries functionSeries = new XYSeries("Function");
+        double unit = (rightCompartment - leftCompartment) / 1000;
+        double minValue = function.apply(leftCompartment);
+        double maxValue = function.apply(leftCompartment);
+        for (double i = leftCompartment; i < rightCompartment; i += unit) {
+            double value = function.apply(i);
+            functionSeries.add(i, value);
+            if (value < minValue) {
+                minValue = value;
+            }
+            if (value > maxValue) {
+                maxValue = value;
+            }
+        }
+
+        XYSeries zeroPositionSeries = new XYSeries("Zero positions");
+        zeroPositionSeries.add(bisectionZeroPosition, function.apply(bisectionZeroPosition));
+        zeroPositionSeries.add(secantZeroPosition, function.apply(secantZeroPosition));
+
+        XYSeries xAxis = new XYSeries("X-axis");
+        xAxis.add(leftCompartment,0);
+        xAxis.add(rightCompartment,0);
+        xAxis.add(0,0);
+
+        XYSeries yAxis = new XYSeries("Y-axis");
```

Comment:
This should be added only if left and right compartments has different signs!

Reply:
If you look to main we don't get chart without different signs.